deploy-board/deploy_board/webapp/cluster_view.py

```diff
@@ -709,10 +710,15 @@ def clone_cluster(request, src_name, src_stage):
         return HttpResponse(json.dumps(src_cluster_info), content_type="application/json")
     except NotAuthorizedException as e:
         log.error("Have an NotAuthorizedException error {}".format(e))
+        if external_id is not None:
+            environs_helper.delete_nimbus_identifier(request, external_id)
```

Comment:
If there is any bad data, will this delete the nimbus identifier causing an incident for old stages?

Reply:
looks like external ID can only be set if new, so i think this one is OK